tests/test_fields_mitigate_collision.py

```diff
@@ -27,10 +27,40 @@ class TestMessage(proto.Message):
         spam_ = proto.Field(proto.STRING, number=1)
         eggs = proto.Field(proto.STRING, number=2)
 
+    class TextStream(proto.Message):
+        text_stream = proto.Field(TestMessage, number=1)
+
     obj = TestMessage(spam_="has_spam")
     obj.eggs = "has_eggs"
     assert obj.spam_ == "has_spam"
 
     # Test that `spam` is coerced to `spam_`
     modified_obj = TestMessage({"spam": "has_spam", "eggs": "has_eggs"})
     assert modified_obj.spam_ == "has_spam"
+
+    # Test __setattr__ and __getattr___
+    modified_obj.__setattr__("spam", "no_spam")
+    modified_obj.__getattr__("spam") == "no_spam"
+
+    modified_obj.__setattr__("spam_", "yes_spam")
+    modified_obj.__getattr__("spam_") == "yes_spam"
+
+    # Try nested values
+    modified_obj = TextStream(
+        text_stream=TestMessage({"spam": "has_spam", "eggs": "has_eggs"})
+    )
+    assert modified_obj.text_stream.spam_ == "has_spam"
+
+    # Test __setattr__ and __getattr___
+    modified_obj.text_stream.__setattr__("spam", "no_spam")
+    modified_obj.text_stream.__getattr__("spam") == "no_spam"
+
+    modified_obj.text_stream.__setattr__("spam_", "yes_spam")
+    modified_obj.text_stream.__getattr__("spam_") == "yes_spam"
+
+    with pytest.raises(KeyError):
+        modified_obj.text_stream.__setattr__("key_does_not_exist", "n/a")
```

Comment:
Same here: use the attribute directly.
We should also add a test for `__getattr__`